class-tgm-plugin-activation.php

```diff
@@ -272,6 +276,34 @@ protected function __construct() {
 		}
 
 		/**
+		 * Magic method to (not) set protected properties from outside of this class.
+		 *
+		 * @internal hackedihack... There is a serious bug in v2.3.2 - 2.3.6  where the `menu` property
+		 * is being assigned rather than tested in a conditional, effectively rendering it useless.
+		 * This 'hack' prevents this from happening.
+		 *
+		 * @see https://github.com/TGMPA/TGM-Plugin-Activation/blob/2.3.6/tgm-plugin-activation/class-tgm-plugin-activation.php#L1593
+		 *
+		 * @param string $name  Name of an inaccessible property.
+		 * @param mixed  $value Value to assign to the property.
+		 * @return void  Silently fail to set the property when this is tried from outside of this class context.
+		 *               (Inside this class context, the __set() method if not used as there is direct access.)
+		 */
+		public function __set( $name, $value ) {
+			return;
```

Comment:
I considered that, but figured the risk is low as development will probably move towards 3.0 now and 3.0 will look quite differently internally.

Anyway, the magic method `__set()` (and `__get()` for that matter) is only called for protected and private properties and only when someone tries to set those from outside the class. Methods within the class have direct access. Same goes for public properties from outside the class (direct access).

In other words, these methods will currently only be called for `$sort_order`, `$has_forced_activation`, `$has_forced_deactivation` and now `$menu`.
The first three should not be changed outside the class anyway (though could be from a child class, aka `protected` rather than `private`).